File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);

    // Print how many quarters to give customer - peso added
    printf("Quarters to give: %i\n", quarters);

    // Print how many quarters to give customer - peso added
    printf("Dimes to give: %i\n", dimes);

    // Print how many quarters to give customer - peso added
    printf("Nickels to give: %i\n", nickels);

    // Print how many quarters to give customer - peso added
    printf("Pennies to give: %i\n", pennies);
}

int get_cents(void)
{
    int n;
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 0 || n > 1000);
    return n;
}

int calculate_quarters(int cents)
{
    // TODO
    int n;
    n = cents / 25;
    return n;
}

int calculate_dimes(int cents)
{
    // TODO
    int n;
    n = cents / 10;
    return n;
}

int calculate_nickels(int cents)
{
    // TODO
    int n;
    n = cents / 5;
    return n;
}

int calculate_pennies(int cents)
{
    // TODO
    int n;
    n = cents / 1;
    return n;
}

File: problem-sets/pset6/sentimental-cash/cash.py
# TODO

from cs50 import get_float


def main():

    # Ask how many cents the customer is owed
    dollars = get_dollars()

    # Calculate the number of quarters to give the customer
    quarters = calculate_quarters(dollars)
    dollars = round(dollars - quarters * 0.25, 2)

    # Calculate the number of dimes to give the customer
    dimes = calculate_dimes(dollars)
    dollars = round(dollars - dimes * 0.10, 2)

    # Calculate the number of nickels to give the customer
    nickels = calculate_nickels(dollars)
    dollars = round(dollars - nickels * 0.05, 2)

    # Calculate the number of pennies to give the customer
    pennies = calculate_pennies(dollars)
    dollars = round(dollars - pennies * 0.01, 2)

    # Sum number of coins
    number_of_coins = quarters + dimes + nickels + pennies

    print(number_of_coins)
    # print("quarters to give: ", quarters)
    # print("dimes to give:    ", dimes)
    # print("nickles to give:  ", nickels)
    # print("pennies  to give: ", pennies)


def get_dollars():
    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            return dollars


def calculate_quarters(dollars):
    return int(dollars / 0.25)


def calculate_dimes(dollars):
    return int(dollars / 0.10)


def calculate_nickels(dollars):
    return int(dollars / 0.05)


def calculate_pennies(dollars):
    return int(dollars / 0.01)


if __name__ == "__main__":
    main()
